reference/format/datagrid.rst

```diff
@@ -0,0 +1,166 @@
+
+    You can reuse existing datagrid configurations by passing their name here.
+
+``filters`` (**type**: ``map``)
+
+    This option currently only provides the ``columns`` key which maps column names to a map which
+    configures how a particular column can be filtered. Available options to configure how to
+    filter a column are:
+
+    ``data_name`` (**type**: ``string``)
+
+        The name of the column from the data source whose values will be filtered.
+
+    ``type`` (**type**: ``string``)
+
+        The type of the filter to be used in the UI.
+
+``options`` (**type**: ``map``)
+
+    Available additional options are:
+
+    ``base_datagrid_class`` (**type**: ``string`` **default**: ``Oro\Bundle\DatagridBundle\Datagrid\Datagrid``)
+
+        With this option, you can switch the datagrid class to a custom implementation.
+
+    ``export`` (**type**: ``boolean`` **default**: ``false``)
+
+        When enabled, the user can export the datagrid in CSV format.
+
+``properties`` (**type**: ``map``)
+
+    This is used for two things: configure how to determine the id of each row and configure links
+    that will be reused in the actions.
+
+    ``id`` (**type**: ``string`` **default**: ``~``)
+
+        The name of the property that acts as an identifier for each entry. By default, the ``id``
+        property is assumed.
+
+    To configure links, use a unique string as an identifier and pass it a map with the following
+    options:
+
+    ``callable`` (**type**: ``string``)
+
+        An expression that will be evaluated when the link is generated.
+
+    ``params`` (**type**: ``map``)
+
+        Additional parameters that are passed to the URL generator together with the configured
+        route name.
+
+    ``route`` (**type**: ``string``)
+
+        The name of the route to the controller action that should be called.
+
+    ``type`` (**type**: ``string``)
+
+        Can be either ``route`` or ``callable`` to use a statically configured route or to
+        dynamically generate a link.
+
+``sorters`` (**type**: ``map``)
+
+    The only available key is ``columns`` which is the map that contains an entry for each column
+    the user can sort the grid by. Each key is the name of a column and the value is a map with
+    the key ``data_name`` mapped to the data source column that will be used to sort the grid.
+
+``source`` (**type**: ``map``)
+
+    The data source that fetches the data to be shown in the grid. Several options control how data
+    are fetched:
+
+    ``aclResource`` (**type**: ``string``)
+
+        An access control list the user must be granted access to in order to actually fetch any
+        data.
+
+    ``type``  (**type**: ``string``)
+
+        The type of data source. Currently, the only available types are ``orm`` and ``search``,
+        but you can also implement your own data source. Each data source may come with its own
+        options to configure how the data is fetched.
```

Comment:
There are quite a lot of datagrid configuration features that are not described here (mass actions, pagination options, entity pagination etc). Not sure that it should contain all of them, because they probably will not be used by 95% of developers.

I'd propose at least add couple of words about entity pagination - it's quite simple and quite useful feature. Here is its description https://github.com/orocrm/platform/blob/a8fa84fef766d44b984a2a40454f316a04199c25/src/Oro/Bundle/EntityPaginationBundle/README.md